Tasks/AzureCLI/azureclitask.ts

```diff
@@ -0,0 +1,197 @@
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import fs = require('fs');
+import util = require('util');
+
+export class azureclitask {
+    public static async runMain() {
+
+        var toolExecutionError = null;
+        try {
+            tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+            var bash = tl.createToolRunner(tl.which('bash', true));
+
+            var scriptPath = tl.getPathInput('scriptPath', true, true);
+            var args = tl.getInput('args', false);
+            var cwd = tl.getPathInput('cwd', true, false);
+            // if user didn't supply a cwd (advanced), then set cwd to folder script is in.
+            // All "script" tasks should do this
+            if (!tl.filePathSupplied('cwd')) {
+                cwd = path.dirname(scriptPath);
+            }
+
+            // determines whether output to stderr will fail a task.
+            // some tools write progress and other warnings to stderr.  scripts can also redirect.
+            var failOnStdErr = tl.getBoolInput('failOnStandardError', false);
+
+            tl.mkdirP(cwd);
+            tl.cd(cwd);
+
+            var connectedServiceNameSelector = tl.getInput('connectedServiceNameSelector', true);
+            this.loginAzure(connectedServiceNameSelector);
+
+            bash.pathArg(scriptPath);
+            bash.argString(args); // additional args should always call argString.  argString() parses quoted arg strings
+
+            await bash.exec({failOnStdErr: failOnStdErr});
+        }
+        catch (err) {
+            if(err.stderr){
+                toolExecutionError = err.stderr;
+            }
+            else {
+                toolExecutionError = err;
+            }
+            //go to finally and logout of azure and set task result
+        }
+        finally {
+            //Logout of Azure if logged in
+            if (this.isLoggedIn) {
+                this.logoutAzure(connectedServiceNameSelector);
+            }
+
+            //set the task result to either succeeded or failed based on error was thrown or not
+            if (toolExecutionError) {
+                tl.setResult(tl.TaskResult.Failed, tl.loc('ScriptFailed', toolExecutionError));
+            }
+            else {
+                tl.setResult(tl.TaskResult.Succeeded, tl.loc('ScriptReturnCode', 0));
+            }
+        }
+    }
+
+    private static isLoggedIn:boolean = false;
+
+    private static loginAzure(connectedServiceNameSelector:string)
```

Comment:
Looks like your login/logout Azure helpers are good candidates to be in common folder (like VstsAzureHelpers_). The Node variant of Azure helpers that you have written can be consumed by many other tasks as well.